src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,464 @@
+ *
+ * Built on top of the Headless UI Listbox: https://headlessui.dev/react/listbox#basic-example
+ *
+ * You can pass in the <Dropdown.Label>, <Dropdown.Button>, and options
+ * (using <Dropdown.Options> and <Dropdown.Option>) through `children` to
+ * have more control and customization over each aspect of the dropdown's
+ * appearance.
+ *
+ * Alternatively, you can pass all of these in using the `labelText`
+ * (or `aria-label`), `buttonText`, and `options` props if you just want
+ * the default styling and don't need to customize the appearance. If you
+ * pass any of these props in, we expect all of them and no `children`.
+ *
+ * Examples:
+ *
+ * ```
+ * return (
+ *   <Dropdown>
+ *     <Dropdown.Label>Options:</Dropdown.Label>
+ *     <Dropdown.Button>Select</Dropdown.Button>
+ *
+ *     <Dropdown.Options>
+ *       <Dropdown.Option>Option 1</Dropdown.Option>
+ *       <Dropdown.Option>Option 2</Dropdown.Option>
+ *       <Dropdown.Option>Option 3</Dropdown.Option>
+ *     </Dropdown.Options>
+ *   </Dropdown>
+ * );
+ * ```
+ *
+ * ```
+ * return (
+ *   <Dropdown aria-label="Options:">
+ *     <Dropdown.Button>Select</Dropdown.Button>
+ *
+ *     <Dropdown.Options>
+ *       <Dropdown.Option>Option 1</Dropdown.Option>
+ *       <Dropdown.Option>Option 2</Dropdown.Option>
+ *       <Dropdown.Option>Option 3</Dropdown.Option>
+ *     </Dropdown.Options>
+ *   </Dropdown>
+ * );
+ * ```
+ *
+ * ```
+ * const options = [
+ *   {
+ *     key: 'option1',
+ *     label: 'Option 1',
+ *   },
+ *   {
+ *     key: 'option2',
+ *     label: 'Option 2',
+ *   },
+ *   {
+ *     key: 'option3',
+ *     label: 'Option 3',
+ *   },
+ * ];
+ *
+ * return (
+ *   <Dropdown labelText="Options:" buttonText="Select" options={options} />
+ * );
+ * ```
+ *
+ * ```
+ * const options = [
+ *   ...
+ * ];
+ *
+ * return (
+ *   <Dropdown aria-label="Options:" buttonText="Select" options={options} />
+ * );
+ * ```
+ *
+ * For compact variant, add variant="compact" and optionally optionsAlign.
+ *
+ * Examples:
+ *
+ *
+ * ```
+ * return (
+ *   <Dropdown
+ *     aria-label="Options"
+ *     optionsAlign="right"
+ *     variant="compact"
+ *   >
+ *     <Dropdown.Options>
+ *       <Dropdown.Option>Option 1</Dropdown.Option>
+ *       <Dropdown.Option>Option 2</Dropdown.Option>
+ *       <Dropdown.Option>Option 3</Dropdown.Option>
+ *     </Dropdown.Options>
+ *   </Dropdown>
+ * );
+ * ```
+ *
+ * ```
+ * const options = [
+ *   ...
+ * ];
+ *
+ * return (
+ *   <Dropdown
+ *     aria-label="Options"
+ *     options={options}
+ *     optionsAlign="right"
+ *     variant="compact"
+ *   />
+ * );
+ * ```
+ *
+ * For dropdown that differs in button and options menu width, style button
+ * width with className and options menu width with optionsClassName.
+ *
+ * Example:
+ *
+ * ```
+ * const options = [
+ *   ...
+ * ];
+ *
+ * return (
+ *   <Dropdown
+ *     aria-label="Options"
+ *     className="w-60"
```

Comment:
I wonder if we should remove tailwind from the examples since we're not using it anymore even through `traject` still is 🤔

Reply:
Hmmmmm, good question. I'll change the className to something "BEM"-ey to show styling capability via classname passing